### PR TITLE
[CSL-1752] Consider HD address in auxx

### PR DESCRIPTION
**Pull request by pva701**

### CODE REVIEW

**Comment by pva701** [comment]
@gromakovsky review poj

File: auxx/src/Pos/Auxx/Hacks.hs [hunk 1]
```diff
@@ -20,3 +23,10 @@ makePubKeyAddressAuxx :: MonadGState m => PublicKey -> m Address
 makePubKeyAddressAuxx pk = do
     ibea <- IsBootstrapEraAddr <$> gsIsBootstrapEra 0
     return $ makePubKeyAddress ibea pk
+
+-- | Similar to @makePubKeyAddressAuxx@ but create HD public key.
+deriveHDAddressAuxx :: MonadGState m => EncryptedSecretKey -> m Address
+deriveHDAddressAuxx hdwSk = do
+    ibea <- IsBootstrapEraAddr <$> gsIsBootstrapEra 0
+    pure $ fst $ fromMaybe (error "makePubKeyHDAddressAuxx: pass mismatch") $
+        deriveFirstHDAddress ibea emptyPassphrase hdwSk
```
Comment (gromakovsky) [inline auxx/src/Pos/Auxx/Hacks.hs]: See #1748, this function should be moved and changed the same way.

File: auxx/src/Pos/Auxx/Hacks.hs [hunk 2]
```diff
@@ -20,3 +23,10 @@ makePubKeyAddressAuxx :: MonadGState m => PublicKey -> m Address
 makePubKeyAddressAuxx pk = do
     ibea <- IsBootstrapEraAddr <$> gsIsBootstrapEra 0
     return $ makePubKeyAddress ibea pk
+
+-- | Similar to @makePubKeyAddressAuxx@ but create HD public key.
```
Comment (gromakovsky) [inline auxx/src/Pos/Auxx/Hacks.hs]: HD public key? 🤔

File: auxx/src/Command/Tx.hs [hunk 3]
```diff
@@ -192,19 +203,30 @@ send
...
+    let plainAddresses = map (flip makePubKeyAddress curPk . IsBootstrapEraAddr) [False, True]
+    let (hdAddresses, hdSecrets) = unzip $ map
+            (\ibea -> fromMaybe (error "send: pass mismatch") $
+                    deriveFirstHDAddress (IsBootstrapEraAddr ibea) emptyPassphrase skey) [False, True]
+    let allAddresses = hdAddresses ++ plainAddresses
+    let allSecrets = hdSecrets ++ [skey, skey]
+    etx <- withSafeSigners allSecrets (pure emptyPassphrase) $ \signersMB -> runExceptT $ do
+        signers <- signersMB `whenNothing` throwError (toException $ AuxxException "Invalid passphrase")
+        let sigAddr = NE.fromList $ zip signers allAddresses
+        -- BE CAREFUL: We create remain address using our pk, wallent doesn't show such addresses
```
Comment (gromakovsky) [inline auxx/src/Command/Tx.hs]: > wallent

File: auxx/src/Pos/Auxx/Hacks.hs (hunk 2 above)
Comment (pva701) [reply auxx/src/Pos/Auxx/Hacks.hs]: oh, I am wrong, HD address

**Comment by pva701** [comment]
@gromakovsky merged w/ master

**Pull request merged**